### ISSUE TITLE
Screen readers cannot list or move focus to elements outside of the current viewport

### ISSUE THREAD

**Issue by jcollins-disney** [flutter/flutter#160217]
### Steps to reproduce

<p>NVDA, JAWS, and VoiceOver Screen readers cannot list or move focus to elements outside of the current viewport. Any elements that have scrolled out of view or haven't yet been scrolled into view will NOT be reachable via screen reader basic keyboard commands for reading and navigating web content.</p>
<p>Examples:</p>
<div class="table-wrap">

Function | JAWS Shortcut | NVDA Shortcut | VoiceOver Shortcut
-- | -- | -- | --
Next Heading | H, 1-6 for heading level | H, 1-6 for heading level | Control Option Command H
List Headings | Insert F6 | Insert F7 to open Elements list, Select the "Headings" radio button | Control Option U, Press left/right arrow keys until the "Heading" menu is displayed
Next Link | U for unvisited link, V for visited link | K for link, U for unvisited, link V for visited link | Control Option Command L for link, Control Option Command V for visited link
List Links | Insert F7 | Insert F7 to open Elements list, Select the "Links" radio button | Control Option U, Press left/right arrow keys until the "Links" menu is displayed
Next Button | B | B | N/A
List Buttons | Control Insert B | Insert F7 to open Elements list, Select the "Buttons" radio button | Control Option U, Press left/right arrow keys until the "Form Controls" menu is displayed
Next Form Control | F | F | Control Option Command J
Next Table | T | T | Control Option Command T
Next Landmark | R | D | N/A
Next List | L | L | Control Option Command X

</div>

### Expected results

- The "Next" keyboard shortcuts should move the screen reader focus to the next element of that type which exists on the page, not just the elements displayed in the current viewport
- The lists should contain all elements of that type which exist on the page, not just the elements displayed in the current viewport

### Actual results

- The "Next" keyboard shortcuts only move the screen reader focus to the next element of that type which is displayed in the current viewport
- The lists only contain elements of that type which are displayed in the current viewport

### Code sample

N/A

### Flutter Doctor output

`Flutter (Channel beta, 3.27.0-0.1.pre, on macOS 14.5 23F79 darwin-arm64, locale en-US)`

**Comment by darshankawar** [comment]
Thanks for the report @jcollins-disney 
Can you provide `flutter doctor -v` to know on which version are you seeing this behavior ?
Are you referring to windows and macos desktop + web platforms for this ?
Can you also take a look at [this](https://github.com/flutter/flutter/issues/83809) issue and open issues listed in [this](https://github.com/flutter/flutter/issues/83809#issuecomment-2150481194) comment to check if they resemble your case or not ?

**Comment by jcollins-disney** [comment]
@darshankawar:
- I've added the flutter `doctor -v info` to the summary above
- Yes, this is Windows and macOS desktop web
- Issue [83809](https://github.com/flutter/flutter/issues/83809) seems to be unrelated to the issue I've listed on this ticket

**Comment by darshankawar** [comment]
Thanks for the update.

**Comment by chunhtai** [comment]
I think this may be due to insufficient cache extent, but I am not to familiar with desktop screen reader on web. assign to web team for further triage

**Comment by Renzo-Olivares** [comment]
Re-opening since  #165589 was reverted. Working on a fix.
